provider/resources.go

```diff
@@ -2339,8 +2393,15 @@ func Provider() tfbridge.ProviderInfo {
 			"azurerm_site_recovery_fabric":               {Tok: azureResource(azureSiteRecovery, "Fabric")},
 			"azurerm_site_recovery_network_mapping":      {Tok: azureResource(azureSiteRecovery, "NetworkMapping")},
 			"azurerm_site_recovery_protection_container": {Tok: azureResource(azureSiteRecovery, "ProtectionContainer")},
-			"azurerm_site_recovery_replicated_vm":        {Tok: azureResource(azureSiteRecovery, "ReplicatedVM")},
-			"azurerm_site_recovery_replication_policy":   {Tok: azureResource(azureSiteRecovery, "ReplicationPolicy")},
+			"azurerm_site_recovery_replicated_vm": {
+				Tok: azureResource(azureSiteRecovery, "ReplicatedVM"),
+				TransformFromState: func(_ context.Context, pm resource.PropertyMap) (resource.PropertyMap, error) {
+					fixEnumCase(pm, "targetDiskType", "Standard_LRS", "Premium_LRS", "StandardSSD_LRS", "UltraSSD_LRS")
+					fixEnumCase(pm, "targetReplicaDiskType", "Standard_LRS", "Premium_LRS", "StandardSSD_LRS", "UltraSSD_LRS")
```

Comment:
Are these TransformFromState for enums necessary for all enums?

Reply:
Where I was able to add tests, I confirmed that the test failed without the `TransformFromState`, however not all resources were possible to get snapshot tests for, so we just added the migration to be safe. There doesn't seem to be much downside as the fixing function is very quick.